Sources/OrderedSet.swift

```diff
@@ -199,6 +199,27 @@ public struct OrderedSet<T: Hashable> {
         remove(sequencedContents[index].pointee)
     }
     
+    /**
+     Removes all objects that satisfy the given predicate in the ordered set.
+     - parameter    shouldBeRemoved: A closure that takes an object in the ordered set as its argument and returns a Boolean value indicating whether the object should be removed from the ordered set.
+     - returns: A collection of the former index positions of the objects. An index position is not provided for objects that were not found.
+    */
+    @discardableResult
+    public mutating func removeAllObjects(where shuldBeRemoved: (T) -> Bool) -> [Index] {
+        
+        let shouldBeRemovedContents = contents.filter { value, _ in
```

Comment:
You're double iterating in this implementation. Maybe something like reduce would simplify this?

```
return contents.reduce([Element]()) { memo, object in 
    var updatedMemo = memo
    if shouldBeRemoved(object) {
        remove(value)
        updatedMemo.append(object)
    }

    return updatedMemo
}
```

This assumes you return the objects instead of indexes. You'd do `contents.enumerated().reduce` and adjust the object to obtain the index as well while iterating contents.

Reply:
I think so(`double iterating` is big problem), i will fix it